tests/io/test_checksumming_proxy.py

```diff
@@ -0,0 +1,53 @@
+import hashlib
+import unittest
+from io import IOBase, BytesIO
+
+from slicedimage.backends._base import ChecksummingContextManager
+
+
+class TestChecksummingProxy(unittest.TestCase):
+    def test_instanceof(self, test_inp="abcde"):
+        obj = BytesIO(test_inp)
+        with ChecksummingContextManager(obj, None) as fh:
+            self.assertIsInstance(fh, IOBase)
+
+    def test_read(self, test_inp="abcde"):
+        obj = BytesIO(test_inp)
+        with ChecksummingContextManager(obj, None) as fh:
+            self.assertEqual(fh.read(2), "ab")
+            self.assertEqual(fh.read(1), "c")
+            self.assertEqual(fh.read(), "de")
+
+    def test_checksum_read(self, test_inp="abcde"):
+        obj = BytesIO(test_inp)
+        hasher = hashlib.sha256()
+        hasher.update(test_inp)
+        expected_sha256 = hasher.hexdigest()
+
+        with ChecksummingContextManager(obj, expected_sha256) as fh:
+            self.assertEqual(fh.read(2), "ab")
+            self.assertEqual(fh.read(1), "c")
+            self.assertEqual(fh.read(), "de")
+
+    def test_checksum_mismatch(self, test_inp="abcde"):
+        obj = BytesIO(test_inp)
+
+        with self.assertRaises(ValueError):
+            with ChecksummingContextManager(obj, "not-the-right-checksum") as fh:
+                self.assertEqual(fh.read(2), "ab")
```

Comment:
If you want each of these to be tested, they need to be under separate `assertRaises` clauses.

Reply:
I'll clarify the code by removing the inner asserts, but this is ok because this is the same as the correct-checksum test.